### DISCUSSION TITLE
CookieSession stored serverside ?

### DISCUSSION THREAD

**Discussion by pullmann4rent** [remix-run/remix#8524] [Q&A]
Hello

I want to avoid sessions on my Server because it can affect the Performance 

so is cookieSession Stored on the Serverside or only in clientside ?

**Comment by sergiodxa** [comment]
`createCookieSessionStorage` uses the cookie itself to store the data, so the data is stored by the browser and sent back in every request to the server.

Other session storages still use a cookie, but only to save the session ID that's used by the session storage to retrieve the session data from where it's stored, this could be Worker KV, Redis, your DB, an API, etc.
